
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
        <link rel="canonical" href="http://dib-lab.github.io/2020-NSURP/08.taxonomic-discovery-with-sourmash/">
      
      
        <link rel="prev" href="../07.quality-control/">
      
      
        <link rel="next" href="../09.comparing-samples-with-sourmash/">
      
      
      <link rel="icon" href="../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.6.0, mkdocs-material-9.5.27">
    
    
      
        <title>Taxonomic Discovery with Sourmash - dib-lab NSURP project</title>
      
    
    
      <link rel="stylesheet" href="../assets/stylesheets/main.6543a935.min.css">
      
        
        <link rel="stylesheet" href="../assets/stylesheets/palette.06af60db.min.css">
      
      


    
    
      
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
    <script>__md_scope=new URL("..",location),__md_hash=e=>[...e].reduce((e,_)=>(e<<5)-e+_.charCodeAt(0),0),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
      

    
    
    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="deep-purple" data-md-color-accent="indigo">
  
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#taxonomic-discovery-with-sourmash" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

  

<header class="md-header md-header--shadow" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href=".." title="dib-lab NSURP project" class="md-header__button md-logo" aria-label="dib-lab NSURP project" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 3 1 9l11 6 9-4.91V17h2V9M5 13.18v4L12 21l7-3.82v-4L12 17l-7-3.82Z"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            dib-lab NSURP project
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Taxonomic Discovery with Sourmash
            
          </span>
        </div>
      </div>
    </div>
    
      
    
    
    
    
      <label class="md-header__button md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
        
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" title="Clear" aria-label="Clear" tabindex="-1">
          
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" tabindex="0" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list" role="presentation"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        <a href="https://github.com/dib-lab/2020-NSURP" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.5.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    2020-NSURP
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    



<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href=".." title="dib-lab NSURP project" class="md-nav__button md-logo" aria-label="dib-lab NSURP project" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 3 1 9l11 6 9-4.91V17h2V9M5 13.18v4L12 21l7-3.82v-4L12 17l-7-3.82Z"/></svg>

    </a>
    dib-lab NSURP project
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/dib-lab/2020-NSURP" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.5.2 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2024 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    2020-NSURP
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href=".." class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Home
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_2" >
        
          
          <label class="md-nav__link" for="__nav_2" id="__nav_2_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Setup
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_2_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_2">
            <span class="md-nav__icon md-icon"></span>
            Setup
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../01.using-farm/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Setting up Farm
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../02.conda/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Install Conda
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../03.lab-notebook/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Keeping a Lab Notebook
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../04.sourmash-tutorial/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Try Sourmash!
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
  
  
    <li class="md-nav__item">
      <a href="../05.starting-a-work-session/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Starting a Work Session
  </span>
  

      </a>
    </li>
  

    
      
      
  
  
    
  
  
  
    
    
    
    
    <li class="md-nav__item md-nav__item--active md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_6" checked>
        
          
          <label class="md-nav__link" for="__nav_6" id="__nav_6_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    IBD Research Project
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_6_label" aria-expanded="true">
          <label class="md-nav__title" for="__nav_6">
            <span class="md-nav__icon md-icon"></span>
            IBD Research Project
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../06.download-assess-ibd-data/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Download and Visually Assess
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../07.quality-control/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Quality Control
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
    
  
  
  
    <li class="md-nav__item md-nav__item--active">
      
      <input class="md-nav__toggle md-toggle" type="checkbox" id="__toc">
      
      
        
      
      
        <label class="md-nav__link md-nav__link--active" for="__toc">
          
  
  <span class="md-ellipsis">
    Taxonomic Discovery with Sourmash
  </span>
  

          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <a href="./" class="md-nav__link md-nav__link--active">
        
  
  <span class="md-ellipsis">
    Taxonomic Discovery with Sourmash
  </span>
  

      </a>
      
        

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#introduction-to-sourmash" class="md-nav__link">
    <span class="md-ellipsis">
      Introduction to sourmash
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#workspace-setup" class="md-nav__link">
    <span class="md-ellipsis">
      Workspace Setup
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#starting-with-sourmash" class="md-nav__link">
    <span class="md-ellipsis">
      Starting with sourmash
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Starting with sourmash">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#install-sourmash" class="md-nav__link">
    <span class="md-ellipsis">
      Install sourmash
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#what-data-to-use" class="md-nav__link">
    <span class="md-ellipsis">
      What data to use?
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#generate-a-sourmash-signature" class="md-nav__link">
    <span class="md-ellipsis">
      Generate a sourmash signature
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#sourmash-gather" class="md-nav__link">
    <span class="md-ellipsis">
      Sourmash gather
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Sourmash gather">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#download-and-unzip-the-database" class="md-nav__link">
    <span class="md-ellipsis">
      Download and unzip the database:
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#run-sourmash-gather" class="md-nav__link">
    <span class="md-ellipsis">
      Run sourmash gather
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#other-methods-for-taxonomic-discovery-and-classification" class="md-nav__link">
    <span class="md-ellipsis">
      Other Methods for Taxonomic Discovery and Classification
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#detecting-contamination-or-incorrect-data" class="md-nav__link">
    <span class="md-ellipsis">
      Detecting contamination or incorrect data
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#challenge-sourmash-gather" class="md-nav__link">
    <span class="md-ellipsis">
      Challenge: sourmash gather
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Challenge: sourmash gather">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#gather-with-trimmed-data" class="md-nav__link">
    <span class="md-ellipsis">
      Gather with trimmed data
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#gather-at-different-ksizes" class="md-nav__link">
    <span class="md-ellipsis">
      Gather at different ksizes
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#test-gather-parameters" class="md-nav__link">
    <span class="md-ellipsis">
      Test Gather parameters
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Test Gather parameters">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#scaled" class="md-nav__link">
    <span class="md-ellipsis">
      scaled
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#base-pair-threshold-for-matches" class="md-nav__link">
    <span class="md-ellipsis">
      base pair threshold for matches
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
      
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../09.comparing-samples-with-sourmash/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Comparing Samples with Sourmash
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../11.experiment-challenge/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Experiment Challenge
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
      
      
  
  
  
  
    
    
    
    
    <li class="md-nav__item md-nav__item--nested">
      
        
        
        <input class="md-nav__toggle md-toggle " type="checkbox" id="__nav_7" >
        
          
          <label class="md-nav__link" for="__nav_7" id="__nav_7_label" tabindex="0">
            
  
  <span class="md-ellipsis">
    Advanced Modules
  </span>
  

            <span class="md-nav__icon md-icon"></span>
          </label>
        
        <nav class="md-nav" data-md-level="1" aria-labelledby="__nav_7_label" aria-expanded="false">
          <label class="md-nav__title" for="__nav_7">
            <span class="md-nav__icon md-icon"></span>
            Advanced Modules
          </label>
          <ul class="md-nav__list" data-md-scrollfix>
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../10.workflows-and-repeatability/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Automation, Workflows, and Repeatability
  </span>
  

      </a>
    </li>
  

              
            
              
                
  
  
  
  
    <li class="md-nav__item">
      <a href="../12.angus-github/" class="md-nav__link">
        
  
  <span class="md-ellipsis">
    Version Control with Git and GitHub
  </span>
  

      </a>
    </li>
  

              
            
          </ul>
        </nav>
      
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#introduction-to-sourmash" class="md-nav__link">
    <span class="md-ellipsis">
      Introduction to sourmash
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#workspace-setup" class="md-nav__link">
    <span class="md-ellipsis">
      Workspace Setup
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#starting-with-sourmash" class="md-nav__link">
    <span class="md-ellipsis">
      Starting with sourmash
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Starting with sourmash">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#install-sourmash" class="md-nav__link">
    <span class="md-ellipsis">
      Install sourmash
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#what-data-to-use" class="md-nav__link">
    <span class="md-ellipsis">
      What data to use?
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#generate-a-sourmash-signature" class="md-nav__link">
    <span class="md-ellipsis">
      Generate a sourmash signature
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#sourmash-gather" class="md-nav__link">
    <span class="md-ellipsis">
      Sourmash gather
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Sourmash gather">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#download-and-unzip-the-database" class="md-nav__link">
    <span class="md-ellipsis">
      Download and unzip the database:
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#run-sourmash-gather" class="md-nav__link">
    <span class="md-ellipsis">
      Run sourmash gather
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#other-methods-for-taxonomic-discovery-and-classification" class="md-nav__link">
    <span class="md-ellipsis">
      Other Methods for Taxonomic Discovery and Classification
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#detecting-contamination-or-incorrect-data" class="md-nav__link">
    <span class="md-ellipsis">
      Detecting contamination or incorrect data
    </span>
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#challenge-sourmash-gather" class="md-nav__link">
    <span class="md-ellipsis">
      Challenge: sourmash gather
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Challenge: sourmash gather">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#gather-with-trimmed-data" class="md-nav__link">
    <span class="md-ellipsis">
      Gather with trimmed data
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#gather-at-different-ksizes" class="md-nav__link">
    <span class="md-ellipsis">
      Gather at different ksizes
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#test-gather-parameters" class="md-nav__link">
    <span class="md-ellipsis">
      Test Gather parameters
    </span>
  </a>
  
    <nav class="md-nav" aria-label="Test Gather parameters">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#scaled" class="md-nav__link">
    <span class="md-ellipsis">
      scaled
    </span>
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#base-pair-threshold-for-matches" class="md-nav__link">
    <span class="md-ellipsis">
      base pair threshold for matches
    </span>
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          
            <div class="md-content" data-md-component="content">
              <article class="md-content__inner md-typeset">
                
                  


<h1 id="taxonomic-discovery-with-sourmash">Taxonomic Discovery with Sourmash</h1>
<p>Until now, we've performed general pre-processing steps on our sequencing data;
sequence quality analysis and trimming usually occur at the start of any sequencing data analysis pipeline.
Now we will begin performing analysis that makes sense for metagenomic sequencing data. </p>
<p>We are working with publicly-available data, but let's pretend that this is a brand new sample that we just got back from our sequencing core. 
One of the first things we often want to do with new metagenome sequencing samples is figure out their approximate species composition. 
This allows us to tap in to all of the information known about these species and relate our community to existing literature. </p>
<p>We can determine the approximate composition of our sample using <code>sourmash</code>. </p>
<h2 id="introduction-to-sourmash">Introduction to sourmash</h2>
<p>Please read <a href="https://angus.readthedocs.io/en/2019/sourmash.html">this tutorial</a> for an introduction to how sourmash works. </p>
<p>tl;dr (but actually please read it): sourmash breaks nucleotide sequences down into k-mers, systematically subsamples those k-mers into a representative "signature", and then enables searches for those k-mers in databases.
This makes it really fast to make comparisons. Here, we will compare our metagenome sample against a pre-prepared database that contains all microbial sequences in GenBank.</p>
<h2 id="workspace-setup">Workspace Setup</h2>
<p>If you're starting a new work session on FARM, be sure to follow the instructions <a href="../05.starting-a-work-session/">here</a>.
You can just do the part to enter a <code>tmux</code> session, since we'll be using a larger <code>srun</code> session than usual.</p>
<h2 id="starting-with-sourmash">Starting with sourmash</h2>
<p>Sourmash doesn't have a big memory or CPU footprint, and can be run on most laptops. 
Below is a recommended <code>srun</code> command to start an interactive session in which to run the <code>srun</code> commands.</p>
<pre><code>srun -p bmh -J sourmash24 -t 24:00:00 --mem=16gb -c 1 --pty bash
</code></pre>
<h3 id="install-sourmash">Install sourmash</h3>
<p><em>Be sure you've set up conda channels properly, as in the <a href="../02.conda/">Install Conda</a> section</em></p>
<pre><code>conda activate nsurp-env
conda install -y sourmash
</code></pre>
<p>Next, let's create a directory in which to store our sourmash signatures</p>
<pre><code>cd ~/2020-NSURP
mkdir -p sourmash
cd sourmash
</code></pre>
<h2 id="what-data-to-use">What data to use?</h2>
<p>We could run sourmash with our adapter trimmed or k-mer trimmed data.
In fact, doing so would make sourmash faster because there would be fewer k-mers in the sample.</p>
<p>We are currently comparing our sample against a database of trusted DNA sequences, so any k-mers in our sample that contain adapters sequence or errors will not match to the trusted reference sequences in the database.
However, even though we very lightly trimmed our reads, there is a chance that we removed a very low abundance organism that was truly present in the sample.
Given this trade-off, we use often raw reads for reference data comparisons, and quality-controlled reads for all other comparisons.</p>
<h2 id="generate-a-sourmash-signature">Generate a sourmash signature</h2>
<p>Next, let's make sourmash signatures from our reads.</p>
<p>Remember from the <a href="https://angus.readthedocs.io/en/2019/sourmash.html">Quick Insights from Sequencing Data with sourmash</a> tutorial that a k-mer size of 21 is approximately specific at the genus level, a 31 is at the species level, and 51 at the strain level. We will calculate our signature with all three k-mer sizes so we can choose which one we want to use later.</p>
<pre><code>sourmash compute -o CSM7KOJE.raw.sig --merge CSM7KOJE --scaled 2000 -k 21,31,51 --track-abundance ~/2020-NSURP/raw_data/CSM7KOJE_*fastq.gz
</code></pre>
<p>You should see output that looks like this:</p>
<pre><code>== This is sourmash version 3.4.1. ==
== Please cite Brown and Irber (2016), doi:10.21105/joss.00027. ==

setting num_hashes to 0 because --scaled is set
computing signatures for files: /home/ntpierce/2020-NSURP/raw_data/CSM7KOJE_R1.fastq.gz, /home/ntpierce/2020-NSURP/raw_data/CSM7KOJE_R2.fastq.gz
Computing signature for ksizes: [21, 31, 51]
Computing only nucleotide (and not protein) signatures.
Computing a total of 3 signature(s).
Tracking abundance of input k-mers.
... reading sequences from /home/ntpierce/2020-NSURP/raw_data/CSM7KOJE_R1.fastq.gz
... /home/ntpierce/2020-NSURP/raw_data/CSM7KOJE_R1.fastq.gz 9704045 sequences
... reading sequences from /home/ntpierce/2020-NSURP/raw_data/CSM7KOJE_R2.fastq.gz
... /home/ntpierce/2020-NSURP/raw_data/CSM7KOJE_R2.fastq.gz 9704045 sequences
calculated 1 signatures for 19408090 sequences taken from 2 files
saved signature(s) to CSM7KOJE.raw.sig. Note: signature license is CC0.
</code></pre>
<p>The outputs file, <code>CSM7KOJE.raw.sig</code> holds a representative subset of k-mers from our original sample, as well as their abundance information. 
The k-mers are "hashed", or transformed, into numbers to make selecting, storing, and looking up the k-mers more efficient.</p>
<h2 id="sourmash-gather">Sourmash gather</h2>
<p><code>sourmash gather</code> is a method for estimating the taxonomic composition of known sequences in a metagenome.</p>
<p>Please go read through the sourmash documentation on <a href="https://sourmash.readthedocs.io/en/latest/classifying-signatures.html#">Breaking down metagenomic samples with gather and lca</a>.
Check out Appendix A and B in this documentation for a good overview of how sourmash gather works.</p>
<p>Running gather on our IBD samples can give us an idea of the microbes present in each sample.
 <code>gather</code> results provide strain-level specificity to matches in its output -- e.g. all strains that match any sequences (above a threshold) in your metagenome will be reported, along with the percent of each strain that matches. 
This is useful both to estimate the amount of metagenome sample that is known, and to estimate the closest strain relative to the organisms in your metagenomes.</p>
<h3 id="download-and-unzip-the-database">Download and unzip the database:</h3>
<pre><code>mkdir -p ~/2020-NSURP/databases/
cd ~/2020-NSURP/databases/
curl -L https://osf.io/jgu93/download -o genbank-k31.sbt.zip
cd ~/2020-NSURP/sourmash
</code></pre>
<h3 id="run-sourmash-gather">Run sourmash gather</h3>
<p>First, let's run a very quick search:</p>
<pre><code>sourmash gather --num-results 10 CSM7KOJE.raw.sig ~/2020-NSURP/databases/genbank-k31.sbt.zip
</code></pre>
<blockquote>
<ul>
<li>the <code>--num-results 10</code> is a way of shortening the search. In this case, we ask for only the top 10 results</li>
</ul>
</blockquote>
<p>We see an output that looks like this:</p>
<pre><code>== This is sourmash version 3.4.1. ==
== Please cite Brown and Irber (2016), doi:10.21105/joss.00027. ==

selecting default query k=31.
loaded query: CSM7KOJE... (k=31, DNA)
loaded 1 databases.


overlap     p_query p_match avg_abund
---------   ------- ------- ---------
6.4 Mbp        9.5%   63.6%      17.4    CBWF010000001.1 Klebsiella pneumoniae...
5.3 Mbp       25.6%   90.6%      56.5    KB851045.1 Clostridium clostridioform...
5.1 Mbp        3.6%   74.4%       8.5    GG668320.1 Clostridium hathewayi DSM ...
4.4 Mbp        2.0%   83.9%       5.4    LBDB01000001.1 Vibrio parahaemolyticu...
3.2 Mbp        5.0%   80.5%      18.6    JTBP01000001.1 Proteus mirabilis stra...
4.8 Mbp        1.0%   33.6%       4.2    JRSL01000930.1 Escherichia coli strai...
2.7 Mbp        0.8%   59.5%       3.5    FUNQ01000052.1 Clostridioides diffici...
2.5 Mbp        3.5%   33.9%      16.3    CZAT01000001.1 Flavonifractor plautii...
4.9 Mbp        2.3%   45.1%      11.7    KQ087951.1 Escherichia coli strain BI...
2.2 Mbp        3.3%   64.6%      18.1    FCEY01000001.1 Clostridium sp. AT5 ge...

found 10 matches total;
(truncated gather because --num-results=10)
the recovered matches hit 56.8% of the query
</code></pre>
<p>The shortened search will be quite quick. </p>
<p>The two columns to pay attention to are <code>p_query</code> and <code>p_match</code>.
<code>p_query</code> is the percent of the metagenome sample that is
(estimated to be) from the named organism.  <code>p_match</code> is the percent
of the database match that is found in the query.<br />
These metrics` are affected by both evolutionary distance and by low coverage of the
organism's gene set (low sequencing coverage, or little expression).</p>
<p>Now, let's run the full <code>gather</code> analysis:
This will take a long time to run.
Sourmash will also output a csv with all the results information that we will use later to visualize our results.</p>
<pre><code>sourmash gather -o CSM7KOJE_x_genbank-k31.gather.csv CSM7KOJE.raw.sig ~/2020-NSURP/databases/genbank-k31.sbt.zip
</code></pre>
<p>When sourmash is finished running, it tells us the % of our sequence was unclassified; i.e. it doesn't match any sequence in the database.</p>
<p>In a later module, we may use additional steps prior to <code>gather</code> to improve the percent of sequence in the metagenome that is classifiable.
These include, for example, using <code>bbduk</code> to remove additional human genome k-mers or using assembly-style programs such as <code>megahit</code> or <a href="https://link.springer.com/article/10.1186/s13059-020-02066-4">spacegraphcats</a>, to build longer contiguous gene sequences.</p>
<h2 id="other-methods-for-taxonomic-discovery-and-classification">Other Methods for Taxonomic Discovery and Classification</h2>
<p>There are many tools, such as Kraken and Kaiju, that can do taxonomic classification of individual reads from metagenomes.
These seem to perform well (albeit with high false positive rates) in situations where you donâ€™t necessarily have the genome sequences that are in the metagenome. 
Sourmash, by contrast, can estimate which known genomes are actually present, so that you can extract them and map/align to them. 
It seems to have a very low false positive rate and is quite sensitive to strains.</p>
<h2 id="detecting-contamination-or-incorrect-data">Detecting contamination or incorrect data</h2>
<p>sourmash <code>gather</code> taxonomic discovery can help uncover contamination or errors in your sequencing samples.
We recommend doing sourmash gather immediately after receiving your data from the sequencing facility.
If your environmental metagenome has a tremendous amount of mouse sequence in it... maybe the sequencing facility sent you the wrong data?</p>
<h2 id="challenge-sourmash-gather">Challenge: sourmash gather</h2>
<h3 id="gather-with-trimmed-data">Gather with trimmed data</h3>
<p>Above, we ran <code>sourmash gather</code> on our untrimmed data. 
44% of the sample did not contain sequence in any GenBank assembly. 
A substantial proportion of this sequence could be due to k-mers with errors.
Run <code>sourmash gather</code> again on the adapter/ k-mer trimmed data.
How much less of the sequence is unclassifiable when the errors and adapters are removed?
How many species are no longer detected after k-mer and error trimming?</p>
<h3 id="gather-at-different-ksizes">Gather at different ksizes</h3>
<p>The genbank reference databases for signatures of ksize k=21 and k=51 are available for download.</p>
<p>k=21</p>
<pre><code>cd ~/2020-NSURP/databases/
curl -L https://osf.io/dm7n4/download -o genbank-k21.sbt.zip
</code></pre>
<p>k=51</p>
<pre><code>cd ~/2020-NSURP/databases/
curl -L https://osf.io/2uvsc/download -o genbank-k51.sbt.zip
</code></pre>
<p>How do you expect the gather results to differ for each? Why?</p>
<h3 id="test-gather-parameters">Test Gather parameters</h3>
<p>By running <code>sourmash gather --help</code>, you can see all the options for the <code>gather</code> program.</p>
<h4 id="scaled">scaled</h4>
<p>The <code>scaled</code> option provides a chaces to downample the query to the specified scaled factor.</p>
<pre><code>  --scaled FLOAT        downsample query to the specified scaled factor
</code></pre>
<p>Try running gather with a <code>scaled</code> value of 50000. How do the results change, and why?</p>
<h4 id="base-pair-threshold-for-matches">base pair threshold for matches</h4>
<p>The <code>threshold-bp</code> option lets you only find matches that have at least this many base pairs in common (default 50,000 bp)</p>
<p>Increasing the threshold makes gather quicker (at the expense of losing some of the smaller matches):</p>
<pre><code>--threshold-bp 10000000
</code></pre>
<p>What happens if you run gather with the threshold above?</p>
<p>Decreasing the threshold will take more time, but be more thorough. A threshold of 0bp does an exhaustive search for all matches</p>
<pre><code>--threshold-bp 0
</code></pre>
<p>The full gather took quite a long time on our samples, so there's no need to run this one!
But do keep it in mind as a way to make sure we get absolutely <strong>all</strong> of the matches we can get using gather.</p>












                
              </article>
            </div>
          
          
<script>var target=document.getElementById(location.hash.slice(1));target&&target.name&&(target.checked=target.name.startsWith("__tabbed_"))</script>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      Copyright &copy; 2020 <a href="http://ivory.idyll.org/lab/">Lab for Data Intensive Biology</a> at UC Davis
    </div>
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    
    
    <script id="__config" type="application/json">{"base": "..", "features": [], "search": "../assets/javascripts/workers/search.b8dbb3d2.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version": "Select version"}}</script>
    
    
      <script src="../assets/javascripts/bundle.ad660dcc.min.js"></script>
      
    
  </body>
</html>